### DISCUSSION TITLE
NvimTree not closing split on nvim-tree.api.tree.close()

### DISCUSSION THREAD

**Discussion by N19htfox** [nvim-tree/nvim-tree.lua#3017] [Q&A]
I'm trying to achieve the following functionality off of a single keybind `<C-e>` for my nvimtree.
When the keybind is pressed, I invoke a function that does the following:
```
If NvimTree window is focused:
    close it
otherwise:
    open it if not opened and focus it
```

I have the script written and mostly working with one problem. When I press my keybind with the NVimTree window focused to close it, it closes nvimtree but not the split. I've even tried:
``` lua
require( 'nvim-tree.api' ).tree.close()
```
``` lua
vim.api.nvim_win_close( tree_split )
```
``` lua
vim.cmd( 'close' )
```

But nothing works. I'm always left with a split where the nvimtree was showing the next available buffer.
How can I ensure the split gets closed as well?
![image](https://github.com/user-attachments/assets/eec619e3-3580-48b5-811f-44639535fe3b)

Then if I press <C-e><C-e>:
![image](https://github.com/user-attachments/assets/7e292bb1-429a-4495-8ecb-d0f82ae35782)

But what I really want is:
![image](https://github.com/user-attachments/assets/d7f41961-3ab6-497a-a376-c998c2ddd0f4)

Is there a proper way to do this or is it a bug?

**Comment by alex-courtis** [comment]
Initial thoughts:

`<C-E>          Open: In Place` is a default binding in the nvim-tree window that could interfere. Please ensure you're unbinding it `:help nvim-tree-mappings`

That if statement can be tricky to evaluate. Idea:
nvim-tree specific mapping (override) to cover the if   (see above)
global mapping to cover the else
